inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -941,6 +941,38 @@ def normalize_collaborations(obj, eng):
     return obj
 
 
+def _get_correct_affiliation_data(affiliations):
+    cleaned_affiliations = []
+    for aff in affiliations:
+        cleaned_affiliations.append({key: val for key, val in aff.items() if key in ['value', 'record']})
+    return cleaned_affiliations
+
+
+def _extract_matched_aff_from_highlight(highlighted_raw_affs, author_raw_affs, author_affs):
+    raw_aff_highlight_lenghts = [len(re.findall('<em>.*?<\/em>', raw_aff)) for raw_aff in highlighted_raw_affs]
```

Comment:
doesn't `len(re.findall(...))` compute the number of pieces? I think you want to find the total matching length instead (without delimiters).

So if you have 
```
highlighted_raw_affs = [ "<em>a</em> string <em>with</em> many <em>short</em> matches", "a string <em>with a much longer single match</em>"]
```
you want to pick the second, not the first, which I believe your code is currently doing (you can add a unit test for that).

Reply:
Currently (I think that it's because `raw_affiliations.value` is a text field) highlighter is splitting it into tokens and then each matched token is highlighted. I couldn't find any solution to change it. 

So the example output for `"authors.raw_affiliations.value": "Institute of Physics Belgrade, Belgrade, Serbia"` is: `"<em>Institute</em> <em>of</em> <em>Physics</em> <em>Belgrade</em>, <em>Belgrade</em>, <em>Serbia</em>"`

Reply:
ok, so you're counting the number of matched tokens. I think total matched length would make more sense though. Do you agree?